packages/components/src/templates/next/components/internal/Vica/VicaWidgetClient.tsx

```diff
@@ -0,0 +1,46 @@
+"use client"
+
+import { useEffect } from "react"
+
+import type { VicaProps } from "~/interfaces"
+
+// Next.js pre-fetching caused widget to disappear on page navigation
+// Doing this forces the widget to load in between page navigation
+const loadVicaScript = () => {
+  const scriptTag = document.createElement("script")
+  scriptTag.async = true
+  scriptTag.type = "text/javascript"
+  scriptTag.src = "https://webchat.vica.gov.sg/static/js/chat.js"
+  scriptTag.referrerPolicy = "origin"
+  document.body.appendChild(scriptTag)
+}
+
+export const VicaWidgetClient = (vica: Omit<VicaProps, "site">) => {
+  useEffect(() => {
+    // to not render during static site generation on the server
+    if (typeof window === "undefined") return
+
+    // Only inject the script after everything else has finished loading
+    // This is to replicate Next.js lazyOnload behaviour (as recommended for widgets)
+    if (document.readyState === "complete") {
+      loadVicaScript()
+    } else {
+      window.addEventListener("load", loadVicaScript)
+      return () => window.removeEventListener("load", loadVicaScript)
```

Comment:
issue: I think we should give the script an ID and remove the script element if it is previously present, similar to the SearchSGInputBox. Otherwise, it can cause multiple scripts to be loaded in the DOM after a couple of route changes.

<img width="926" alt="image" src="https://github.com/user-attachments/assets/40053ee4-dc51-4746-947f-67d9c96f6573" />

Reply:
thanks for catching, updated in https://github.com/opengovsg/isomer/pull/1086/commits/34a185ffa87aaf5d6bfa8465a1f72b5623dbbe9d !